api/python/quilt3/data_transfer.py

```diff
@@ -473,7 +469,10 @@ def _upload_or_copy_file(ctx, size, src_path, dest_bucket, dest_path):
         try:
             params = dict(Bucket=dest_bucket, Key=dest_path)
             s3_client = ctx.s3_client_provider.find_correct_client(S3Api.HEAD_OBJECT, dest_bucket, params)
-            resp = s3_client.head_object(**params)
+            resp = s3_client.get_object_attributes(
```

Comment:
I think you can use `head_object(**params, ChecksumMode='ENABLED')` instead. That way we don't need new permissions. Also `get_object_attributes()` returns checksum without a number of parts at the end and as I understand we want it, BTW do we really want it?

Reply:
Will try `head_bucket`; didn't know about that parameter.

`get_object_attributes` returns parts if the object was uploaded using a multipart upload.

Reply:
> `get_object_attributes` returns parts if the object was uploaded using a multipart upload.

```
In [82]: resp = s3.get_object_attributes(
    ...:     Bucket='quilt-sergey-test-promote',
    ...:     Key='32MB-file',
    ...:     ObjectAttributes=['ObjectParts', 'Checksum'],
    ...: )
    ...: print(resp['Checksum']['ChecksumSHA256'])
    ...: print(resp['ObjectParts']['TotalPartsCount'])
fMJJ08ZgZU9Z8aMET6oXADfcmluSqcXXtR07Xy3DCoQ=
2
```

Reply:
Oh wow, you're right. I somehow missed it, and it looks like a bug in S3 - it's literally the same checksum, just without the number of parts.

But `head_object(..., ChecksumMode='ENABLED')` works correctly - so I'll do that.